test/e2e/compSchemaValidation_test.go

```diff
@@ -0,0 +1,86 @@
+package e2e
+
+import (
+	"testing"
+	"time"
+
+	"sigs.k8s.io/e2e-framework/pkg/features"
+	"sigs.k8s.io/e2e-framework/third_party/helm"
+
+	apiextensionsv1 "github.com/crossplane/crossplane/apis/apiextensions/v1"
+	pkgv1 "github.com/crossplane/crossplane/apis/pkg/v1"
+	"github.com/crossplane/crossplane/test/e2e/config"
+	"github.com/crossplane/crossplane/test/e2e/funcs"
+)
+
+const (
+	// SuiteCompositionWebhookSchemaValidation is the value for the
+	// config.LabelTestSuite label to be assigned to tests that should be part
+	// of the Composition Webhook Schema Validation test suite.
+	SuiteCompositionWebhookSchemaValidation = "composition-webhook-schema-validation"
+)
+
+func init() {
+	E2EConfig.AddTestSuite(SuiteCompositionWebhookSchemaValidation,
+		config.WithHelmInstallOpts(
+			helm.WithArgs("--set args={--debug,--enable-composition-webhook-schema-validation}"),
+		),
+		config.WithLabelsToSelect(features.Labels{
+			config.LabelTestSuite: []string{SuiteCompositionWebhookSchemaValidation, config.TestSuiteDefault},
+		}),
+	)
+}
+
+func TestCompositionValidation(t *testing.T) {
+	manifests := "test/e2e/manifests/apiextensions/composition/validation"
+
+	cases := features.Table{
+		{
+			// A valid Composition should be created when validated in strict mode.
+			Name:        "ValidCompositionIsAcceptedStrictMode",
+			Description: "A valid Composition should be created when validated in strict mode.",
+			Assessment: funcs.AllOf(
+				funcs.ApplyResources(FieldManager, manifests, "composition-valid.yaml"),
+				funcs.ResourcesCreatedWithin(30*time.Second, manifests, "composition-valid.yaml"),
+			),
+		},
+		{
+			// An invalid Composition should be rejected when validated in strict mode.
+			Name:       "InvalidCompositionIsRejectedStrictMode",
+			Assessment: funcs.ResourcesFailToApply(FieldManager, manifests, "composition-invalid.yaml"),
+		},
+	}
+	environment.Test(t,
+		cases.Build("CompositionValidation").
+			WithLabel(LabelStage, LabelStageAlpha).
+			WithLabel(LabelArea, LabelAreaAPIExtensions).
+			WithLabel(LabelSize, LabelSizeSmall).
+			WithLabel(LabelModifyCrossplaneInstallation, LabelModifyCrossplaneInstallationTrue).
+			WithLabel(config.LabelTestSuite, SuiteCompositionWebhookSchemaValidation).
+			// Enable our feature flag.
+			WithSetup("EnableAlphaCompositionValidation", funcs.AllOf(
+				funcs.AsFeaturesFunc(funcs.HelmUpgrade(E2EConfig.GetSuiteInstallOpts(SuiteCompositionWebhookSchemaValidation)...)),
+				funcs.ReadyToTestWithin(1*time.Minute, namespace),
```

Comment:
we need to be explicit here to ensure it works in all the ways the test can be selected and called (labels, presets, go test selectors), otherwise we risk having to do conditionals again as I'm already doing for the `additionalSetupFuncs`. However, we can simplify this line which is going to be in many test cases by adding a function to E2EConfig to upgrade crossplane with some configuration, something like:
```suggestion
				E2EConfig.UpgradeCrossplaneForSuite(SuiteCompositionWebhookSchemaValidation),
				funcs.ReadyToTestWithin(1*time.Minute, namespace),
```